### ISSUE TITLE
Fails to install tweaked IPA from Azule: Guru Meditation e6aca1@147:a45007

### ISSUE THREAD

**Issue by yodaluca23** [Al4ise/Azule#144]
ERROR: Guru Meditation e6aca1@147:a45007 could not find executable for C:\Users\user\AppData\Local\Temp\tmpr14574yh\Payload\Spotify.app\Frameworks\CydiaSubstrate.framework Install failed: Guru Meditation e6aca1@147:a45007 could not find executable for C:\Users\user\AppData\Local\Temp\tmpr14574yh\Payload\Spotify.app\Frameworks\CydiaSubstrate.framework

I injected a couple of tweaks, into an IPA, with Azule, but when I try to sideload it with Sideloadly, I get the above error... Also, if I unzip the IPA, and navigate to the Frameworks/CydiaSubstrate.framework their is a CydiaSubstrate file.

**Comment by asdfzxcvbn** [comment]
yep the `CydiaSubstrate` binary is inside the .framework, maybe your antivirus is deleting the framework? i've never personally encountered this issue

**Comment by yodaluca23** [comment]
> yep the `CydiaSubstrate` binary is inside the .framework, maybe your antivirus is deleting the framework? i've never personally encountered this issue

I tried doing it from AppDB, so that the anti-virus wouldn't be a problem, and it installs, but gives the integrity verification error...

![image](https://user-images.githubusercontent.com/67206487/235550773-6059139e-6236-4fbb-9bbc-6178995be867.jpeg)

**Comment by yur1xpp** [comment]
@Al4ise Do you have any idea why? I gets the same issue. I did some testing by manually extracting the injecting libraries `CydiaSubstrate.framework` created by Sideloadly IPA, then use the `-m` flag in Azule to inject this framework instead. Now, it sideloaded successfully. The differences in the frameworks: (left=Sideloadly, right=Azule)

![image](https://github.com/Al4ise/Azule/assets/34767893/769e1cdd-b199-4af1-a56c-29b382fa4ae1)

**Comment by asdfzxcvbn** [comment]
> @Al4ise Do you have any idea why? I gets the same issue. I did some testing by manually extracting the injecting libraries `CydiaSubstrate.framework` created by Sideloadly IPA, then use the `-m` flag in Azule to inject this framework instead. Now, it sideloaded successfully. I tested sideloading using [AltServer-Linux](https://github.com/NyaMisty/AltServer-Linux) rather than Sideloadly itself to verify it's not Sideloadly specific issue. The differences in the frameworks: (left=Sideloadly, right=Azule)
> 
> ![image](https://user-images.githubusercontent.com/34767893/265194257-769e1cdd-b199-4af1-a56c-29b382fa4ae1.png)

i remember this issue. use azule's `CydiaSubstrate.framework`, but remove the Info.plist and make a new one with the following contents:

```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
    <key>CFBundleDevelopmentRegion</key>
    <string>English</string>
    <key>CFBundleExecutable</key>
    <string>CydiaSubstrate</string>
    <key>CFBundleGetInfoString</key>
    <string>Cydia Substrate, SaurikIT LLC</string>
    <key>CFBundleIdentifier</key>
    <string>com.saurik.CydiaSubstate</string>
    <key>CFBundleInfoDictionaryVersion</key>
    <string>6.0</string>
    <key>CFBundleName</key>
    <string>Cydia Substrate</string>
    <key>CFBundlePackageType</key>
    <string>FMWK</string>
    <key>CFBundleShortVersionString</key>
    <string>0.9</string>
    <key>CFBundleSignature</key>
    <string>????</string>
</dict>
</plist>
```

this should fix everything.

**Comment by asdfzxcvbn** [comment]
> I got the same issue and interestingly, it does indeed fixed the issue! Looks like the one in Azule is erroneous, might be during parsing process on linux (Ubuntu). I don't think the end quote and semicolon should be there. At least iOS version is not there. (I'm on Ubuntu, x86_64) ![image](https://user-images.githubusercontent.com/45226508/265207711-333f9f64-dc26-49b0-b693-0e61a17d4085.png)
> 
> cc @Al4ise since this might be a bug

i have my own injector and i figured this out while fixing trollstore compatibility. as far as i know, only two valid plist formats exist: binary and xml. the `Info.plist` in azule's `CydiaSubstrate.framework` is incorrectly formatted and can't be parsed by most apps. that's why using xml format fixes it.

**Comment by yur1xpp** [comment]
> > @Al4ise Do you have any idea why? I gets the same issue. I did some testing by manually extracting the injecting libraries `CydiaSubstrate.framework` created by Sideloadly IPA, then use the `-m` flag in Azule to inject this framework instead. Now, it sideloaded successfully. I tested sideloading using [AltServer-Linux](https://github.com/NyaMisty/AltServer-Linux) rather than Sideloadly itself to verify it's not Sideloadly specific issue. The differences in the frameworks: (left=Sideloadly, right=Azule)
> > ![image](https://user-images.githubusercontent.com/34767893/265194257-769e1cdd-b199-4af1-a56c-29b382fa4ae1.png)
> 
> i remember this issue. use azule's `CydiaSubstrate.framework`, but remove the Info.plist and make a new one with the following contents:
> 
> ```
> <?xml version="1.0" encoding="UTF-8"?>
> <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
> <plist version="1.0">
> <dict>
>     <key>CFBundleDevelopmentRegion</key>
>     <string>English</string>
>     <key>CFBundleExecutable</key>
>     <string>CydiaSubstrate</string>
>     <key>CFBundleGetInfoString</key>
>     <string>Cydia Substrate, SaurikIT LLC</string>
>     <key>CFBundleIdentifier</key>
>     <string>com.saurik.CydiaSubstate</string>
>     <key>CFBundleInfoDictionaryVersion</key>
>     <string>6.0</string>
>     <key>CFBundleName</key>
>     <string>Cydia Substrate</string>
>     <key>CFBundlePackageType</key>
>     <string>FMWK</string>
>     <key>CFBundleShortVersionString</key>
>     <string>0.9</string>
>     <key>CFBundleSignature</key>
>     <string>????</string>
> </dict>
> </plist>
> ```
> 
> this should fix everything.

Can confirm this works for me too! Thanks man!
